src/main/java/io/cdap/plugin/gcp/dataplex/common/config/DataplexBaseConfig.java

```diff
@@ -0,0 +1,244 @@
+      if (credentials == null) {
+        failureCollector.addFailure(String.format("Unable to load credentials from %s.",
+              connection.isServiceAccountFilePath() ? connection.getServiceAccountFilePath() : "provided JSON key"),
+            "Ensure the service account file is available on the local filesystem.")
+          .withConfigProperty("serviceFilePath")
+          .withConfigProperty("serviceAccountJSON");
+        throw failureCollector.getOrThrowException();
+      }
+    }
+  }
+
+  private GoogleCredentials getCredentialsFromServiceAccount() throws IOException {
+    GoogleCredentials credentials = null;
+    // validate service account
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      credentials =
+        GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath())
+          .createScoped(Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
+    } else {
+      credentials = ServiceAccountCredentials.getApplicationDefault().createScoped(
+        Lists.newArrayList(DataplexConstants.DATAPLEX_SCOPE));
+    }
+    return credentials;
+  }
+
+  protected void configureDataplexException(String dataplexConfigProperty, String dataplexConfigPropType,
+                                            ApiException e,
+                                            FailureCollector failureCollector) {
+    if (StatusCode.Code.NOT_FOUND.equals(e.getStatusCode().getCode()) ||
+      StatusCode.Code.INVALID_ARGUMENT.equals(e.getStatusCode().getCode())) {
+      failureCollector
+        .addFailure("'" + dataplexConfigProperty + "' could not be found. Please ensure that it exists in " +
+          "Dataplex.", null).withConfigProperty(dataplexConfigPropType);
+    } else if (StatusCode.Code.PERMISSION_DENIED.equals(e.getStatusCode().getCode()) ||
+      StatusCode.Code.UNAUTHENTICATED.equals(e.getStatusCode().getCode())) {
+      failureCollector
+        .addFailure("'" + dataplexConfigProperty + "' could not be accessed. Please ensure that you have" +
+          " required permissions.", null).withConfigProperty(dataplexConfigPropType);
+    } else {
+      failureCollector.addFailure(e.getCause().getMessage(), null)
+        .withConfigProperty(dataplexConfigPropType);
+    }
+
+  }
+
+  @Nullable
+  public String getServiceAccountFilePath() {
+    return connection == null ? null : connection.getServiceAccountFilePath();
+  }
+
+  /**
+   * Service account email is required to create task in dataplex.
+   *
+   * @return service account email ID
+   * @throws IOException
+   */
+  public String getServiceAccountEmail() throws IOException {
+    if (connection.isServiceAccountJson() || connection.getServiceAccountFilePath() != null) {
+      ServiceAccountCredentials credentials =
+        GCPUtils.loadServiceAccountCredentials(getServiceAccount(), isServiceAccountFilePath());
+      return credentials.getClientEmail();
+    } else {
```

Comment:
else part is for auto-detect in project and filepath field.

Reply:
You already return in the previous if condition, so you don't need else here

Reply:
in case of service account json, it will go to if loop and return that email and in case of auto-detect it will go to else block and will fetch the email from there.